sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -364,6 +364,21 @@ class QueryCompilationErrorsSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("INVALID_JSON_SCHEMA_MAPTYPE: " +
+    "Parse JSON rows can only contain StringType as a key type for a MapType.") {
+    withTempPath { path =>
+      val schema = StructType(
+        StructField("map", MapType(IntegerType, IntegerType, true), false) :: Nil)
+      val e = intercept[AnalysisException] {
+        spark.read.schema(schema).json(path.getCanonicalPath)
+      }
+      assert(e.getErrorClass === "INVALID_JSON_SCHEMA_MAPTYPE")
+      assert(e.getMessage ===
+        "Input schema StructType(StructField(map,MapType(IntegerType,IntegerType,true),false)) " +
+          "can only contain StringType as a key type for a MapType.")
```

Comment:
Could you use the `checkErrorClass` method added by https://github.com/apache/spark/pull/36287.